### PR TITLE
40 seed factory

### PR DESCRIPTION
**Pull request by kinspotting**
Closes https://github.com/PrimeDAO/contracts/issues/40

### CODE REVIEW

**Comment by mfahampshire** [comment]
`Duplicate contract names found for SafeERC20` warning on compilation

File: contracts/seed/Seed.sol [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+    */
+    function buy(uint256 _amount) public protected checked {
+        require((fundingToken.balanceOf(address(this)) + _amount) <= cap, "Seed: amount exceeds contract sale cap");
+        require(fundingToken.transferFrom(msg.sender, address(this), _amount), "Seed: no tokens");
+
+        if (fundingToken.balanceOf(address(this)) >= successMinimum) {
+            minimumReached = true;
+        }
+
+        fundingTokensPerAddress[msg.sender] = _amount;
```
Comment (kinspotting) [inline contracts/seed/Seed.sol:165]: why don't we store it in the struct?

File: contracts/seed/Seed.sol [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+
+        uint feeAmount = (_amount.mul(uint(PPM))).mul(fee).div(100000000); // 100000000 = PPM * 100
```
Comment (kinspotting) [inline contracts/seed/Seed.sol:167]: can you please move this mafic value to the variable? maybe we can call it PPM100 or PPMBASE

File: contracts/seed/Seed.sol [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+        fees[msg.sender] = feeAmount;
```
Comment (kinspotting) [inline contracts/seed/Seed.sol:168]: maybe could be moved to the struct as well

File: contracts/seed/Seed.sol [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+        Lock storage tokenLock = tokenLocks[_locker];
+        tokenLock.daysClaimed = uint16(tokenLock.daysClaimed.add(daysVested));
+        tokenLock.totalClaimed = uint256(tokenLock.totalClaimed.add(amountVested));
+
+        require(seedToken.transfer(beneficiary, fees[_locker]), "Seed: cannot transfer to beneficiary");
+        require(seedToken.transfer(tokenLock.recipient, amountVested), "Seed: no tokens");
+        emit TokensClaimed(tokenLock.recipient, amountVested);
+    }
+
+    // returns funding tokens to user
```
Comment (kinspotting) [inline contracts/seed/Seed.sol:193]: /**
      * @dev                            Returns funding tokens to user.
    */

**Review by kinspotting — COMMENTED** [review]
pls see the comments

**Review by mfahampshire — APPROVED** [review]

**Pull request merged**